pkg/repositories/models/execution.go

```diff
@@ -56,4 +56,10 @@ type Execution struct {
 	// The user responsible for launching this execution.
 	// This is also stored in the spec but promoted as a column for filtering.
 	User string `gorm:"index" valid:"length(0|255)"`
+	// GORM doesn't save the zero value for ints, so we use a pointer for the State field
+	State *int32 `gorm:"default:0"`
+	// State updated at to indicate the archival/activation time
+	StateUpdatedAt *time.Time
+	// The user responsible for activating/archiving this execution.
+	StateUpdatedBy string `gorm:"index" valid:"length(0|255)"`
```

Comment:
why does this need an index?

Reply:
I think if we have time range queries  to get set of executions then this might be useful, but we can always decide this based on the usecases and then add them later. I have removed it for now

Reply:
sure sounds good! i don't really anticipate we'll want to query on state changes immediately but this will be okay to not have backfilled if we change our minds